### PR TITLE
a tiny js world oop

### PR DESCRIPTION
**Pull request by madmaxWMFU**
### A Tiny JS World OOP

[Demo](https://madmaxwmfu.github.io/a-tiny-JS-world/) |
[Code base](https://github.com/madmaxwmfu/a-tiny-JS-world/blob/gh-pages/index.js)

Please, review.

### CODE REVIEW

File: submissions/madmaxWMFU/a-tiny-JS-world-OOP/index.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+    allAbout() {
+        let allAbout = super.allAbout();
+        let firstPart = allAbout.slice(0, 3);
+        let secondPart = allAbout.slice(3, 4);
+        return [...firstPart, this.wings, this.paws, ...secondPart].join("; ");
+    }    
+}
+
+const man = new Human('Max', 'human', 'male', 2, 2, 'Hallo!');
```
Comment (o-msh) [inline submissions/madmaxWMFU/a-tiny-JS-world-OOP/index.js]: No need to define species of human here. We can define this property in constructor. Also don't forget about default properties.

File: submissions/madmaxWMFU/a-tiny-JS-world-OOP/index.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+class Objects {
```
Comment (o-msh) [inline submissions/madmaxWMFU/a-tiny-JS-world-OOP/index.js]: We must define classes with singular, but `Object` name is reserved. Maybe use another common name? `Inhabitant` will be better.

**Comment by madmaxWMFU** [comment]
I completely forgot that `Objects `have a method `toStrnig()`
Thanks for sending me the right way!
I corrected all the shortcomings.

File: submissions/madmaxWMFU/a-tiny-JS-world-OOP/index.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            this.species,
+            `<strong>${this.name}</strong>`,
+            this.gender,
+            this.hands,
+            this.legs,
+            `<em>${this.say()}</em>`
+        ].join("; ");        
+    }
+
+    checkValue(checkNum, constNum) {
```
Comment (o-msh) [inline submissions/madmaxWMFU/a-tiny-JS-world-OOP/index.js]: This is redundant method. You can use `||` for setting default values.

File: submissions/madmaxWMFU/a-tiny-JS-world-OOP/index.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        return typeof(checkNum) != "undefined" ? checkNum : constNum;
+    }
+}
+
+class Human extends Inhabitant {
+    constructor(props) {
+        super(props);
+        this.hands = this.checkValue(props.hands, 2);
```
Comment (o-msh) [inline submissions/madmaxWMFU/a-tiny-JS-world-OOP/index.js]:
```javascript
this.hands = prop.hands || ...
```

**Pull request merged**